Tella/Data/Database/TellaDataBase.swift

```diff
@@ -675,5 +637,119 @@ extension TellaDataBase {
+        guard let locale = serversDict.first else { return nil }
+        return try self.parseDicToObjectOf(type: UwaziLocale.self, dic: locale)
+    }
+    func getAllUwaziLocale() throws -> [UwaziLocale] {
+        let serversDict = try statementBuilder.selectQuery(tableName: D.tUwaziServerLanguage,
+                                                           andCondition: [])
+        if !serversDict.isEmpty {
+            return try self.parseDicToObjectOf(type: [UwaziLocale].self, dic: serversDict)
+        }
+        return []
+    }
+
+    func deleteUwaziLocaleWith(serverId : Int) throws {
+        statementBuilder.delete(tableName: D.tUwaziServerLanguage,
+                                primarykeyValue: [KeyValue(key: D.cServerId, value: serverId)])
+    }
+
+    func deleteAllUwaziLocale() throws -> Int {
+        return try statementBuilder.deleteAll(tableNames: [D.tUwaziServerLanguage])
+    }
+}
+extension TellaDataBase: UwaziTemplateProtocol {
+    func createTemplateTableForUwazi() {
+        let columns = [
+            cddl(D.cId, D.integer, primaryKey: true, autoIncrement: true),
+            cddl(D.cTemplateId, D.text),
+            cddl(D.cTemplateEntity, D.text),
+            cddl(D.cTemplateDownloaded, D.integer),
+            cddl(D.cTemplateUpdated, D.integer),
+            cddl(D.cTemplateFavorited, D.integer),
+            cddl(D.cServerName, D.text),
+            cddl(D.cServerId, D.integer, tableName: D.tServer, referenceKey: D.cServerId)
+
+        ]
+        statementBuilder.createTable(tableName: D.tUwaziTemplate, columns: columns)
+    }
+    func getUwaziTemplateeWith(serverId: Int) throws -> CollectedTemplate? {
+        let serversDict = try statementBuilder.selectQuery(tableName: D.tUwaziTemplate,
+                                                           andCondition: [KeyValue(key: D.cServerId, value: serverId)])
+        guard let locale = serversDict.first else { return nil }
+        return try self.parseDicToObjectOf(type: CollectedTemplate.self, dic: locale)
+    }
+    func getUwaziTemplateeWith(templateId: Int) throws -> CollectedTemplate? {
+        let serversDict = try statementBuilder.selectQuery(tableName: D.tUwaziTemplate,
+                                                           andCondition: [KeyValue(key: D.cTemplateId, value: templateId)])
+        guard let locale = serversDict.first else { return nil }
+        return try self.parseDicToObjectOf(type: CollectedTemplate.self, dic: locale)
+    }
+    func getAllUwaziTemplate() throws -> [CollectedTemplate] {
+        let serversDict = try statementBuilder.selectQuery(tableName: D.tUwaziTemplate,
+                                                           andCondition: [])
+        if !serversDict.isEmpty {
+            return try self.parseDicToObjectOf(type: [CollectedTemplate].self, dic: serversDict)
+        }
+        return []
+    }
+    func addUwaziTemplateWith(template: CollectedTemplate) throws -> CollectedTemplate {
```

Comment:
why are we using the suffix "with" at the end of every method? shouldn't be more simple `addUwaziTemplate`?

Reply:
Ya that make scene I will remove it 
We used to add preposition to end of the function to make it more human readable.